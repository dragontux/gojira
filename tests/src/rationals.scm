; check for correctness at most basic level
(write (/ 1 2))
(newline)

(write (/ 10 100))
(newline)

; check for lexer correctness
(write 1/2)
(newline)

(write 30/40)
(newline)

(write -1/2)
(newline)

(write 4/-5)
(newline)

; check addition
(write (+ 2/3 1/3))
(newline)

(write (+ 2/4 1/3 1))
(newline)

(write (+ -2/4 1/3))
(newline)

; check subtraction
(write (- 4 1/3))
(newline)

(write (- 6/4 1/2))
(newline)

(write (- -6/4 1/2))
(newline)

; check multiplication
(write (* 1/3 1/3))
(newline)

(write (* 4 -1/4))
(newline)

(write (* -1/2 2/5))
(newline)

; check division
(write (/ 1/3 1/3))
(newline)

(write (/ 1 1/4))
(newline)

(write (/ 5 10))
(newline)
