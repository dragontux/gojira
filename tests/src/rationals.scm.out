1/2
1/10
1/2
3/4
-1/2
-4/5
1
11/6
-1/6
11/3
1
-2
1/9
-1
-1/5
1
4
1/2
